### PR TITLE
Instructor: edit session: multiple choice questions: support generating options for all other teams #8661

### LINKED ISSUE #8661: Instructor: edit session: multiple choice questions: support generating options for all other teams
Similar to #8412 but for 'all other teams' instead of 'all other students'

### PR DESCRIPTION
**Pull request by dalessr**
Fixes #8661 

**Outline of Solution**
- Add a new option `teams (excluding self)`
- Check the team name of this participant so that the team can be removed from `teamList`
- Add tests to check the correctness of the display of corresponding teams

**Before:**
37624952-e34e42a8-2beb-11e8-8a78-266e4d8ae51e

**After:**
8661-after-1

8661-after-2

### CODE REVIEW

**Comment by dalessr** [comment]
Status: Ready for review

File: src/test/java/teammates/test/cases/browsertests/StudentFeedbackSubmitPageUiTest.java [hunk 1]
```diff
@@ -613,6 +613,18 @@ private void testInputValidation() {
...
         submitPage.verifyAndCloseSuccessfulSubmissionModal();
         submitPage.waitForTextsForAllStatusMessagesToUserEquals(Const.StatusMessages.FEEDBACK_RESPONSES_SAVED);
+
+        ______TS("cannot choose self when generating choices from teams (excluding self)");
+        submitPage = loginToStudentFeedbackSubmitPage("Alice", "Open Session");
+        assertFalse(submitPage.checkIfMcqOrMsqChoiceExists(26, 0, "Team 1"));
+        assertTrue(submitPage.checkIfMcqOrMsqChoiceExists(26, 0, "Team 2"));
+
+        assertFalse(submitPage.checkIfMcqOrMsqChoiceExists(27, 0, "Team 1"));
+        assertTrue(submitPage.checkIfMcqOrMsqChoiceExists(27, 0, "Team 2"));
```
[the remark this answers is not in the document]
Comment (dalessr) [reply src/test/java/teammates/test/cases/browsertests/StudentFeedbackSubmitPageUiTest.java]: @Haozhe321 Based on my understanding of the method `checkIfMcqOrMsqChoiceExists`, it only checks whether the certain option exists in the question, but we cannot ensure that multiple options can be selected?
For example, if I add another selection (Team 3) here and `assertTrue`, I can just say that both options (Team 2 & Team 3) exist in the option list, but I cannot guarantee that both can be selected?

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java [hunk 2]
```diff
@@ -338,15 +342,34 @@ public String getQuestionWithoutExistingResponseSubmissionFormHtml(
...
      */
     private int generateNumOfChoicesForStudentsExcludingSelf(String courseId) {
         List<StudentAttributes> studentList = StudentsLogic.inst().getStudentsForCourse(courseId);
 
         return studentList.size() - 1;
     }
 
+    /**
+     * Gets the number of MSQ choices for teams excluding self.
+     * Returns zero if the course does not exist.
```
[the remark this answers is not in the document]
Comment (dalessr) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java]: Here my concern is that, even if the assertion will fail, I still need to return something for this method to show that the `courseId` does not exist. Or is it a better way if I just throw the error up to the `setMsqQuestionDetails` method and show the error message there without actually updating the `numOfMsqChoices` attribute?

File: src/test/java/teammates/test/cases/browsertests/StudentFeedbackSubmitPageUiTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Haozhe321) [reply src/test/java/teammates/test/cases/browsertests/StudentFeedbackSubmitPageUiTest.java]: That's true, nvm about it then.

File: src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Haozhe321) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java]: If it goes into the catch block due to an `EntityDoesNotExistException`, the page will be redirected to a `500 INTERNAL SERVER ERROR` page, the so the return at the end is just so the method can compile.

[the remark this answers is not in the document]
Comment (dalessr) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java]: Yes that is true. So is it ok if I just modify the header comment?

[the remark this answers is not in the document]
Comment (Haozhe321) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java]: Removing this line `Returns zero if the course does not exist.` will do.

[the remark this answers is not in the document]
Comment (dalessr) [reply src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java]: Okay noted, thanks!

**Comment by dalessr** [comment]
Status: Ready for review

**Comment by Haozhe321** [comment]
LGTM

**Pull request merged**